packages/tools/devtools/devtools-browser-extension/src/content/ContentScriptContent.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	type ISourcedDevtoolsMessage,
+	devtoolsMessageSource,
+	isDevtoolsMessage,
+} from "@fluidframework/devtools-core/internal";
+
+import {
+	extensionPopupMessageSource,
+	extensionViewMessageSource,
+	relayMessageToPort,
+} from "../messaging/index.js";
+
+import {
+	contentScriptMessageLoggingOptions,
+	formatContentScriptMessageForLogging,
+} from "./Logging.js";
+
+/**
+ * Code run when "ContentScript" executes.
+ * Factored out for testability.
+ */
+export function runContentScript(
+	browser: typeof chrome,
+	window: (Window & typeof globalThis) | undefined,
+): void {
+	type Port = chrome.runtime.Port;
+
+	/**
+	 * This module is the extension's Content Script.
+	 * It lives in the tab context, alongside the page being communicated with.
+	 *
+	 * The lifetime of the script itself is roughly the same as the lifetime of the tab, but in our case it
+	 * doesn't do anything until it is activated by the Background Worker.
+	 *
+	 * Once initialized, this script relays messages between the tab and the Background Worker, which in turn communicates
+	 * with the Devtools extension.
+	 *
+	 * For an overview of how the various scripts communicate in the Devtools extension model,
+	 * see {@link https://developer.chrome.com/docs/extensions/mv3/devtools/#content-script-to-devtools | here}.
+	 */
+
+	console.log(formatContentScriptMessageForLogging("Initializing Content Script."));
+
+	// `window` should always be defined in the Content script context.
+	if (window === undefined) {
+		throw new Error("Window object is not defined.");
+	}
+
+	/* eslint-disable @typescript-eslint/no-non-null-assertion */
```

Comment:
Nit: seems like this disable is not necessary anymore and can be removed together with line 110.